### PR TITLE
Correctly unmarshal Protocol

### PR DESCRIPTION
**Pull request by ph**
The Protocol (https/http) was not correctly unmarshaly and assigned to
the value of the struct. Because of this the checking was trying to use
a plain text connection or a TLS enabled endpoint.

Fixes: #17841

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by ph** [comment]
@michalpristas Lets make sure all our test in CI for integration only uses TLS.

**Comment by ph** [comment]
I can confirm this fixes the issues in cloud.

File: x-pack/elastic-agent/CHANGELOG.asciidoc [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - Remove the kbn-version on each request to the Kibana API. {pull}17764[17764]
 - Fixed process spawning on Windows {pull}17751[17751]
 - Fixed injected log path to monitoring beat {pull}17833[17833]
+- Make sure that the Elastic Agent connect over TLS in cloud. {pull}xx[xxx]
```
Comment (ruflin) [inline x-pack/elastic-agent/CHANGELOG.asciidoc:26]: @ph Can you fix this in a follow up?

**Comment by ruflin** [comment]
Merged this already to get it into the build cycle.

File: x-pack/elastic-agent/CHANGELOG.asciidoc (hunk 1 above)
Comment (ph) [reply x-pack/elastic-agent/CHANGELOG.asciidoc:26]: :facepalm: one day we wont need this.

**Pull request merged**